### PR TITLE
fix: added a test for errors inside promise

### PR DESCRIPTION
**Pull request by yury-s**
Also bump Chromium revision past 686227 which contained v8 roll with the actual fix.

References #4651

### CODE REVIEW

**Comment by yury-s** [comment]
It seems that the new release will break a bunch of tests, after inspecting CSP changes with Andrey I skipped one of them. There are also crashes in the browser that cause worker tests to timeout:

Failures:
1) Chromium Browser Page Target should create a worker from a service worker (target.spec.js:86:5)
  Message:
    Timeout Exceeded 10000ms

2) Chromium Browser Page Target should create a worker from a shared worker (target.spec.js:93:5)
  Message:
    Timeout Exceeded 10000ms

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "node": ">=6.4.0"
   },
   "puppeteer": {
-    "chromium_revision": "681777"
+    "chromium_revision": "686378"
```
Comment (JoelEinbinder) [inline package.json]: we generally do rolls as their own commits, as they can lots of unforeseen side effects.

File: test/evaluation.spec.js [hunk 2]
```diff
@@ -255,6 +255,13 @@ module.exports.addTests = function({testRunner, expect}) {
       const a = await page.evaluate(() => Array(100 * 1024 * 1024 + 1).join('a'));
       expect(a.length).toBe(100 * 1024 * 1024);
     });
+    it('should throw error with detailed information on exception inside promise ', async({page, server}) => {
+      let error = 'did not catch error';
+      await page.evaluate(() => new Promise(() => {
+        does_not_exist.click();
```
Comment (JoelEinbinder) [inline test/evaluation.spec.js]: i think this test would be easier to read if you just throw an error here

File: test/evaluation.spec.js [hunk 3]
```diff
@@ -255,6 +255,13 @@ module.exports.addTests = function({testRunner, expect}) {
       const a = await page.evaluate(() => Array(100 * 1024 * 1024 + 1).join('a'));
       expect(a.length).toBe(100 * 1024 * 1024);
     });
+    it('should throw error with detailed information on exception inside promise ', async({page, server}) => {
+      let error = 'did not catch error';
```
Comment (JoelEinbinder) [inline test/evaluation.spec.js]: nit: `let error = null` to be consistent with our other tests that look at errors

File: test/evaluation.spec.js (hunk 2 above)
Comment (yury-s) [reply test/evaluation.spec.js]: Done.

File: test/evaluation.spec.js (hunk 3 above)
Comment (yury-s) [reply test/evaluation.spec.js]: I find error messages like 
'Cannot read property 'message' of null'
less informative than
'expect.toContain failed: did not catch error ⊇ Error in promise'

but I agree that the approach should be consistent across the codebase, so changing it to null, we can update the tests later.

**Comment by yury-s** [comment]
This is now blocked on #4841

**Review by aslushnikov — APPROVED** [review]
